src/services/configServices/SettingsService.js

```diff
@@ -167,6 +168,11 @@ class SettingsService {
     }
     const { id, isPrivate } = siteInfo.value
     if (!isPrivate && !enablePassword) return okAsync("")
+    if (!isPrivate && enablePassword) {
+      // Additionally, we perform an async operation here to swap the netlify repo to private and disable builds,
+      // only when changing from public to private
+      privatiseNetlifySite(siteName, password)
```

Comment:
can we add a comment on why we are not awaiting this

Reply:
Added in 0ec15298!